backend/composer/services/graph_service.py

```diff
@@ -157,17 +155,55 @@ def compile_journey(connectivity_statement) -> dict:
     # Extract origins, vias, and destinations from the connectivity statement
     Via = apps.get_model('composer', 'Via')
     Destination = apps.get_model('composer', 'Destination')
+    Origin = apps.get_model('composer', 'AnatomicalEntity')
 
-    origins = list(connectivity_statement.origins.all())
-
-    vias = list(Via.objects.filter(connectivity_statement=connectivity_statement))
-    destinations = list(Destination.objects.filter(connectivity_statement=connectivity_statement))
+    vias = list(Via.objects.filter(connectivity_statement__id=connectivity_statement.id))
+    destinations = list(Destination.objects.filter(connectivity_statement__id=connectivity_statement.id))
+    origins = list(Origin.objects.filter(origins_relations__id=connectivity_statement.id).distinct())
+    # origins = list(connectivity_statement.origins.all())
 
     # Generate all paths and then consolidate them
     all_paths2 = generate_paths(origins, vias, destinations)
-    consolidated_paths, journey_paths = consolidate_paths(all_paths2)
+    consolidated_paths = consolidate_paths(all_paths2)
+    return consolidated_paths
+
+
+def get_journey_path_from_consolidated_paths(consolidated_paths):
+    journey_paths = [[((node[1].replace(JOURNEY_DELIMITER, ' or '), node[2]) if (
+        node[2] == 0 or path.index(node) == len(path) - 1) else (
+        node[1].replace(JOURNEY_DELIMITER, ', '), node[2])) for node in path] for path in consolidated_paths]
+    return journey_paths
+
+
+def build_journey_description(consolidated_paths, connectivity_statement):
+    journey_paths = get_journey_path_from_consolidated_paths(
+        consolidated_paths)
+    Via = apps.get_model('composer', 'Via')
+    vias = list(Via.objects.filter(
```

Comment:
question: Why do we need to fetch the vias here? can't we rely entirely on the new journey paths attribute created?

Reply:
I created this card to handle this at a later stage:
https://metacell.atlassian.net/browse/SCKAN-390